### PR TITLE
MM-20535 Convert create_team to TypeScript

### PR DESCRIPTION
**Pull request by josephk96**
#### Summary
I converted ```components/create_team.jsx``` and associated files and tests to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13706
Jira: https://mattermost.atlassian.net/browse/MM-20535

### CODE REVIEW

File: components/create_team/create_team.tsx [hunk 1]
```diff
@@ -1,68 +1,76 @@
...
+    };
 
     render() {
         const {
             currentChannel,
             currentTeam,
             customDescriptionText,
             match,
-            siteName,
+            siteName
```
Comment (deanwhillier) [inline components/create_team/create_team.tsx]: Looks like our eslint rules are a bit out of whack at the moment. As far as I'm aware, we expect a trailing comma here like so:
```suggestion
            siteName,
```

**Comment by deanwhillier** [comment]
@josephk96, could you please do a sync and merge on this when you have a minute?

**Comment by josephk96** [comment]
> @josephk96, could you please do a sync and merge on this when you have a minute?

Done!

**Comment by hanzei** [comment]
Thanks for the contribution @josephk96 :tada:

**Pull request merged**